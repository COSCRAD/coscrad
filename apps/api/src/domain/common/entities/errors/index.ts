export * from './cannot-add-duplicate-translation.error';
