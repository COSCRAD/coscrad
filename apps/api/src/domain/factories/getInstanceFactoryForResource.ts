import { DomainModelCtor } from '../../lib/types/DomainModelCtor';
import { ResultOrError } from '../../types/ResultOrError';
import { Resource } from '../models/resource.entity';
import { ResourceType } from '../types/ResourceType';
import buildBibliographicReferenceFactory from './complexFactories/buildBibliographicReferenceFactory';
import buildSpatialFeatureFactory from './complexFactories/buildSpatialFeatureFactory';
import buildInstanceFactory from './utilities/buildInstanceFactory';
import getAggregateCtorFromAggregateType from './utilities/getAggregateCtorFromAggregateType';

export type InstanceFactory<TResourceType> = (dto: unknown) => ResultOrError<TResourceType>;

/**
 * It would be nice to find a pattern that gives us better type safety.
 */
export default <TResource extends Resource>(
    resourceType: ResourceType
): InstanceFactory<TResource> => {
    if (resourceType === ResourceType.spatialFeature)
        // @ts-expect-error TODO fix this tricky type error
        return buildSpatialFeatureFactory();

    if (resourceType === ResourceType.bibliographicReference)
        // @ts-expect-error TODO fix this tricky type error
        return buildBibliographicReferenceFactory();

    const Ctor = getAggregateCtorFromAggregateType(resourceType);

    return buildInstanceFactory<TResource>(Ctor as DomainModelCtor<TResource>);
};
