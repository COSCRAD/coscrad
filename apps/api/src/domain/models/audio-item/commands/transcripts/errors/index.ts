export * from './cannot-add-inconsistent-line-item.error';
export * from './failed-to-import-line-items-to-transcript.error';
export * from './invalid-timestamp-order.error';
export * from './invalid-transcript.error';
