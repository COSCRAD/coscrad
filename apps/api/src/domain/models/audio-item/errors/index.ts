export * from './CannotOverwriteTranscriptError';
export * from './conflicting-line-items.error';
export * from './DuplicateTranscriptParticipantErrors';
export * from './DuplicateTranscriptParticipantInitialsError';
export * from './DuplicateTranscriptParticipantNameError';
export * from './line-item-out-of-bounds.error';
export * from './transcript-line-item';
