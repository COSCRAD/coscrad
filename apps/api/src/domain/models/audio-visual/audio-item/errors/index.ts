export * from '../../shared/transcript-errors/cannot-overwrite-transcript-error';
export * from '../../shared/transcript-errors/conflicting-line-items.error';
export * from '../../shared/transcript-errors/duplicate-transcript-participant-errors';
export * from '../../shared/transcript-errors/duplicate-transcript-participant-initials-error';
export * from '../../shared/transcript-errors/duplicate-transcript-participant-name-error';
export * from '../../shared/transcript-errors/line-item-out-of-bounds.error';
