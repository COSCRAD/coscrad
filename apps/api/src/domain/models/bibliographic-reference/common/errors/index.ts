export * from './cannot-override-digital-representation.error';
