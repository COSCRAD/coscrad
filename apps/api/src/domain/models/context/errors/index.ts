export * from './cannot-add-audio-for-note-in-given-language.error';
