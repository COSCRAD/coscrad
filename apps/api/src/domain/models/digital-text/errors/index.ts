export * from './cannot-add-audio-for-missing-content.error';
export * from './cannot-add-page-with-duplicate-identifier.error';
export * from './cannot-add-photograph-for-missing-page.error';
export * from './cannot-import-pages-to-non-empty-digital-text.error';
export * from './cannot-override-cover-photograph-error';
export * from './cannot-overwrite-page-content.error';
export * from './cannot-overwrite-photograph-for-page.error';
export * from './duplicate-digital-text-title.error';
export * from './empty-page-import.error';
export * from './failed-to-import-pages-to-digital-text.error';
export * from './failed-to-update-digital-text-page.error';
export * from './missing-original-text-item-in-page-import.error';
export * from './missing-page.error';
export * from './multiple-original-items-in-page-import.error';
