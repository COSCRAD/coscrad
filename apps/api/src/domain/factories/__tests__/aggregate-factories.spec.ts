import { InternalError } from '../../../lib/errors/InternalError';
import formatAggregateType from '../../../queries/presentation/formatAggregateType';
import { DTO } from '../../../types/DTO';
import { Aggregate } from '../../models/aggregate.entity';
import { isResourceType, ResourceType } from '../../types/ResourceType';
import getInstanceFactoryForResource from '../getInstanceFactoryForResource';
import buildAggregateFactoryTestCases from './buildAggregateFactoryTestCases';

const testCaseSets = buildAggregateFactoryTestCases();

describe(`Aggregate factories`, () => {
    /**
     * TODO [https://www.pivotaltracker.com/story/show/183109452]
     * Change this to `AggregateType` and add test coverage for non-resource
     * aggregates.
     */
    Object.values(ResourceType).forEach((aggregateType) =>
        describe(`An aggregate of type: ${formatAggregateType(aggregateType)}`, () => {
            describe(`the corresponding test suite`, () => {
                const testSuite = testCaseSets.find(
                    ({ aggregateType: testSuiteAggregateType }) =>
                        aggregateType === testSuiteAggregateType
                );

                it(`should exist`, () => {
                    expect(testSuite).toBeTruthy();
                });

                it(`should have at least one valid test case`, () => {
                    expect(testSuite?.validCases?.length || 0).toBeGreaterThan(0);
                });

                it(`should have at least one invalid test case`, () => {
                    expect(testSuite?.invalidCases?.length || 0).toBeGreaterThan(0);
                });
            });
        })
    );

    testCaseSets
        // TODO [https://www.pivotaltracker.com/story/show/183109452] Remove filter
        .filter(({ aggregateType }) => isResourceType(aggregateType))
        .forEach(({ aggregateType, validCases, invalidCases }) => {
            // TODO [https://www.pivotaltracker.com/story/show/183109452] remove cast when removing filter
            const buildInstance = getInstanceFactoryForResource(aggregateType as ResourceType);

            describe(`when attempting to build an instance of type: ${aggregateType} from a DTO`, () => {
                describe(`when the DTO is valid`, () => {
                    validCases.forEach(({ dto, description }) =>
                        describe(description, () => {
                            it('should succeed', () => {
                                const result = buildInstance(dto);

                                expect(result).not.toBeInstanceOf(InternalError);

                                // Verify the mapping from DTO to instance by inspection
                                expect(result).toMatchSnapshot();
                            });
                        })
                    );
                });

                invalidCases.forEach(({ description, dto, checkError }) =>
                    describe(description, () => {
                        it('should return an appropriate error', () => {
                            const result = buildInstance(dto as DTO<Aggregate>);

                            checkError(result);
                        });
                    })
                );
            });
        });
});
