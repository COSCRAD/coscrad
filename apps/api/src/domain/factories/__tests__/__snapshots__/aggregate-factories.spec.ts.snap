// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aggregate factories when attempting to build an instance of type: audioItem from a DTO when the DTO is valid valid transribed audio DTO should succeed 1`] = `
WithTranscription {
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110110",
  "lengthMilliseconds": 20000,
  "mediaItemId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "name": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "en",
        "role": "original",
        "text": "The Wooden Boy",
      },
    ],
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "transcript": Transcript {
    "items": [
      TranscriptItem {
        "inPointMilliseconds": 12000,
        "outPointMilliseconds": 15550,
        "speakerInitials": "JHC",
        "text": MultilingualText {
          "items": [
            MultilingualTextItem {
              "languageCode": "en",
              "role": "original",
              "text": "There once was a little wooden boy.",
            },
          ],
        },
      },
      TranscriptItem {
        "inPointMilliseconds": 18300,
        "outPointMilliseconds": 19240,
        "speakerInitials": "JHC",
        "text": MultilingualText {
          "items": [
            MultilingualTextItem {
              "languageCode": "en",
              "role": "original",
              "text": "His name was Pinocchio",
            },
          ],
        },
      },
    ],
    "participants": [
      TranscriptParticipant {
        "initials": "JHC",
        "name": "Jimmy H. Cricket",
      },
    ],
  },
  "type": "audioItem",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: bibliographicReference from a DTO when the DTO is valid valid dto for a Bibliographic Reference of sub-type: Book should succeed 1`] = `
BookBibliographicReference {
  "data": BookBibliographicReferenceData {
    "abstract": "This is the abstract, NOT a general note!",
    "creators": [
      {
        "name": "Alana Duvernay",
        "type": "author",
      },
      {
        "name": "James Smith",
        "type": "author",
      },
    ],
    "isbn": "978-1-895811-34-6",
    "numberOfPages": 455,
    "place": "Kennebunk Port, Maine",
    "publisher": "Atlantic Publishing",
    "title": "A Day in the Life",
    "type": "book",
    "url": "https://coscrad.org/wp-content/uploads/2023/05/mock-book-bibliographic-reference-1_a-day-in-the-life.pdf",
    "year": 1999,
  },
  "digitalRepresentationResourceCompositeIdentifier": {
    "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b100001",
    "type": "digitalText",
  },
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "bibliographicReference",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: bibliographicReference from a DTO when the DTO is valid valid dto for a Bibliographic Reference of sub-type: Court Case should succeed 1`] = `
CourtCaseBibliographicReference {
  "data": CourtCaseBibliographicReferenceData {
    "abstract": "Doe, Jane
- Large Valley
- Qualifying as an expert in the written language",
    "caseName": "2002-07-08_NationTitleCase",
    "court": "Supreme Court of British Columbia",
    "dateDecided": "Recorded 2002-07-08",
    "pages": "Pages 1-6",
    "type": "courtCase",
    "url": "https://coscrad.org/wp-content/uploads/2023/05/mock-court-case-bibliographic-reference-3_title-case.pdf",
  },
  "digitalRepresentationResourceCompositeIdentifier": undefined,
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110003",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "bibliographicReference",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: bibliographicReference from a DTO when the DTO is valid valid dto for a Bibliographic Reference of sub-type: Journal Article should succeed 1`] = `
JournalArticleBibliographicReference {
  "data": JournalArticleBibliographicReferenceData {
    "abstract": "An analysis of the Cariboo Chilcotin Justice inquiry.",
    "creators": [
      BibliographicReferenceCreator {
        "name": "Sigurd Purcell",
        "type": "author",
      },
    ],
    "doi": "10.14288/bcs.v0i19.784",
    "issn": "00052949",
    "issueDate": "Spring 2013",
    "publicationTitle": "Journal of History",
    "title": "Report on the Cariboo Chilcotin Justice inquiry",
    "type": "journalArticle",
    "url": "https://coscrad.org/wp-content/uploads/2023/05/Test-Journal-Article-Sample-PDF.pdf",
  },
  "digitalRepresentationResourceCompositeIdentifier": undefined,
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110023",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "bibliographicReference",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: book from a DTO when the DTO is valid valid book should succeed 1`] = `
Book {
  "author": "John Doeburg",
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110023",
  "pages": [
    BookPage {
      "identifier": "1",
      "text": "foo",
      "translation": "dee foo",
    },
    BookPage {
      "identifier": "2",
      "text": "blah blah blah",
      "translation": "blahas",
    },
  ],
  "publicationDate": "2002",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "subtitle": "The wolf's revenge",
  "title": "Three Little Pigs",
  "type": "book",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: digitalText from a DTO when the DTO is valid valid digital text should succeed 1`] = `
DigitalText {
  "eventHistory": [
    {
      "meta": {
        "dateCreated": 1664237194999,
        "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b100950",
        "userId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b100685",
      },
      "payload": {
        "aggregateCompositeIdentifier": {
          "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
          "type": "digitalText",
        },
        "languageCodeForTitle": "hai",
        "title": "Court Case Digital Text title in language",
      },
      "type": "DIGITAL_TEXT_CREATED",
    },
  ],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "pages": [
    DigitalTextPage {
      "content": null,
      "identifier": "1",
    },
  ],
  "published": false,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "title": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "hai",
        "role": "original",
        "text": "Court Case Digital Text title in language",
      },
    ],
  },
  "type": "digitalText",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: mediaItem from a DTO when the DTO is valid valid media item should succeed 1`] = `
MediaItem {
  "contributorAndRoles": [
    ContributorAndRole {
      "contributorId": "2",
      "role": "host",
    },
  ],
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "lengthMilliseconds": 2500,
  "mimeType": "audio/x-wav",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "title": "episode (media item 1) title (in language)",
  "titleEnglish": "Metal Mondays episode 1",
  "type": "mediaItem",
  "url": "https://coscrad.org/wp-content/uploads/2023/05/metal-mondays-mock1_533847__tosha73__distortion-guitar-power-chord-e.wav",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: photograph from a DTO when the DTO is valid valid photograph should succeed 1`] = `
Photograph {
  "dimensions": PhotographDimensions {
    "heightPX": 400,
    "widthPX": 300,
  },
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110000",
  "mediaItemId": "5",
  "photographer": "Susie McRealart",
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "title": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "en",
        "role": "original",
        "text": "Adiitsii Running",
      },
    ],
  },
  "type": "photograph",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: playlist from a DTO when the DTO is valid valid playlist DTO should succeed 1`] = `
Playlist {
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110501",
  "items": [
    PlaylistItem {
      "resourceCompositeIdentifier": {
        "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110110",
        "type": "audioItem",
      },
    },
    PlaylistItem {
      "resourceCompositeIdentifier": {
        "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110111",
        "type": "audioItem",
      },
    },
  ],
  "name": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "clc",
        "role": "original",
        "text": "Blake's jams",
      },
    ],
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "playlist",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: song from a DTO when the DTO is valid valid song should succeed 1`] = `
Song {
  "audioURL": "https://coscrad.org/wp-content/uploads/2023/05/mock-song-1_mary-had-a-little-lamb.wav",
  "contributions": [
    ContributorAndRole {
      "contributorId": "1",
      "role": "performer",
    },
  ],
  "eventHistory": [
    {
      "meta": {
        "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110009",
      },
      "payload": {
        "aggregateCompositeIdentifier": {
          "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
          "type": "song",
        },
        "audioURL": "https://coscrad.org/wp-content/uploads/2023/05/mock-song-1_mary-had-a-little-lamb.wav",
        "languageCodeForTitle": "clc",
        "title": "Song title in language",
      },
      "type": "SONG_CREATED",
    },
  ],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "lengthMilliseconds": 3500,
  "lyrics": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "en",
        "role": "original",
        "text": "Mary had a little lamb, little lamb.",
      },
    ],
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "startMilliseconds": 0,
  "title": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "clc",
        "role": "original",
        "text": "Song title in language",
      },
      MultilingualTextItem {
        "languageCode": "en",
        "role": "free translation",
        "text": "Mary had a little lamb (English)",
      },
    ],
  },
  "type": "song",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: spatialFeature from a DTO when the DTO is valid valid dto for a Spatial Feature of sub-type: Line should succeed 1`] = `
Line {
  "eventHistory": [],
  "geometry": {
    "coordinates": [
      [
        52.3,
        -124.2,
      ],
      [
        52.35,
        -124.85,
      ],
      [
        52.21,
        -124.9,
      ],
      [
        52.05,
        -125.1,
      ],
      [
        52.15,
        -125.1,
      ],
      [
        52.33,
        -125.22,
      ],
      [
        52.45,
        -125.39,
      ],
    ],
    "type": "LineString",
  },
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "properties": SpatialFeatureProperties {
    "description": "One amazing hike!",
    "imageUrl": "https://coscrad.org/wp-content/uploads/2023/05/wind-energy-2029621_1280.png",
    "name": "Windy Path",
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "spatialFeature",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: spatialFeature from a DTO when the DTO is valid valid dto for a Spatial Feature of sub-type: Point should succeed 1`] = `
Point {
  "eventHistory": [],
  "geometry": {
    "coordinates": [
      54.034404852745006,
      -132.17563335558532,
    ],
    "type": "Point",
  },
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110100",
  "properties": SpatialFeatureProperties {
    "description": "Description for point 100",
    "imageUrl": "https://coscrad.org/wp-content/uploads/2023/05/evergreen-2025158_1280.png",
    "name": "Name of Point with ID: 100",
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "spatialFeature",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: spatialFeature from a DTO when the DTO is valid valid dto for a Spatial Feature of sub-type: Polygon should succeed 1`] = `
Polygon {
  "eventHistory": [],
  "geometry": {
    "coordinates": [
      [
        [
          52.47,
          -123.6,
        ],
        [
          52.42,
          -123.95,
        ],
        [
          52.05,
          -123.95,
        ],
        [
          52.05,
          -123.685,
        ],
      ],
    ],
    "type": "Polygon",
  },
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110300",
  "properties": SpatialFeatureProperties {
    "description": "Description for polygon 300",
    "imageUrl": "https://coscrad.org/wp-content/uploads/2023/05/map-1272165_640.png",
    "name": "Point with ID: 300 Name",
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "spatialFeature",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: term from a DTO when the DTO is valid when the dto is valid should succeed 1`] = `
Term {
  "audioFilename": undefined,
  "contributorId": "John Doe",
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110001",
  "isPromptTerm": false,
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "sourceProject": undefined,
  "text": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "clc",
        "role": "original",
        "text": "Chil-term-1",
      },
      MultilingualTextItem {
        "languageCode": "en",
        "role": "free translation",
        "text": "Engl-term-1",
      },
    ],
  },
  "type": "term",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: video from a DTO when the DTO is valid valid video DTO should succeed 1`] = `
WithTranscription {
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110223",
  "lengthMilliseconds": 20000,
  "mediaItemId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110002",
  "name": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "en",
        "role": "original",
        "text": "The Demonstration",
      },
    ],
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "transcript": Transcript {
    "items": [
      TranscriptItem {
        "inPointMilliseconds": 12000,
        "outPointMilliseconds": 15550,
        "speakerInitials": "DM",
        "text": MultilingualText {
          "items": [
            MultilingualTextItem {
              "languageCode": "en",
              "role": "original",
              "text": "This is how.",
            },
          ],
        },
      },
      TranscriptItem {
        "inPointMilliseconds": 18300,
        "outPointMilliseconds": 19240,
        "speakerInitials": "DM",
        "text": MultilingualText {
          "items": [
            MultilingualTextItem {
              "languageCode": "en",
              "role": "original",
              "text": "It is done",
            },
          ],
        },
      },
    ],
    "participants": [
      TranscriptParticipant {
        "initials": "DM",
        "name": "Dee Monstrator",
      },
    ],
  },
  "type": "video",
}
`;

exports[`Aggregate factories when attempting to build an instance of type: vocabularyList from a DTO when the DTO is valid valid vocabulary list should succeed 1`] = `
VocabularyList {
  "entries": [
    VocabularyListEntry {
      "termId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110511",
      "variableValues": {
        "person": "11",
        "positive": true,
      },
    },
    VocabularyListEntry {
      "termId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110512",
      "variableValues": {
        "person": "12",
        "positive": false,
      },
    },
    VocabularyListEntry {
      "termId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110513",
      "variableValues": {
        "person": "13",
        "positive": false,
      },
    },
    VocabularyListEntry {
      "termId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110501",
      "variableValues": {
        "person": "01",
        "positive": true,
      },
    },
    VocabularyListEntry {
      "termId": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b110502",
      "variableValues": {
        "person": "02",
        "positive": false,
      },
    },
  ],
  "eventHistory": [],
  "getCompositeIdentifier": [Function],
  "id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b114567",
  "name": MultilingualText {
    "items": [
      MultilingualTextItem {
        "languageCode": "clc",
        "role": "original",
        "text": "To Sing (lang)",
      },
      MultilingualTextItem {
        "languageCode": "en",
        "role": "free translation",
        "text": "To Sing (Engl)",
      },
    ],
  },
  "published": true,
  "queryAccessControlList": AccessControlList {
    "allowedGroupIds": [],
    "allowedUserIds": [],
  },
  "type": "vocabularyList",
  "variables": [
    VocabularyListFilterProperty {
      "name": "positive",
      "type": "checkbox",
      "validValues": [
        {
          "display": "negative (lha)",
          "value": false,
        },
        {
          "display": "positive form (switch for negative)",
          "value": true,
        },
      ],
    },
    VocabularyListFilterProperty {
      "name": "person",
      "type": "dropbox",
      "validValues": [
        {
          "display": "I",
          "value": "1",
        },
        {
          "display": "You",
          "value": "2",
        },
        {
          "display": "She",
          "value": "3",
        },
      ],
    },
  ],
}
`;
